### PR TITLE
[WebNN EP] Support Sign and CumSum operators

### PR DESCRIPTION
**Pull request by miaobin**
This PR supports Sign and CumSum operators for WebNN EP. @Honry @fdwr  PTAL, thanks.

### CODE REVIEW

File: onnxruntime/core/providers/webnn/builders/impl/cumsum_op_builder.cc [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  emscripten::val output = emscripten::val::object();
+  output = model_builder.GetBuilder().call<emscripten::val>("cumulativeSum", input, narrow<uint32_t>(axis), options);
+  model_builder.AddOperand(node.OutputDefs()[0]->Name(), std::move(output));
+  return Status::OK();
+}
+
+// Operator support related.
+bool CumSumOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& /* initializers */,
+                                        const Node& node,
+                                        WebnnDeviceType device_type,
```
Comment (Honry) [inline onnxruntime/core/providers/webnn/builders/impl/cumsum_op_builder.cc:63]:
```suggestion
                                        WebnnDeviceType /* device_type */,
```

**Review by Honry — APPROVED** [review]
LGTM % a nit.

File: onnxruntime/core/providers/webnn/builders/impl/cumsum_op_builder.cc (hunk 1 above)
Comment (miaobin) [reply onnxruntime/core/providers/webnn/builders/impl/cumsum_op_builder.cc]: Done.

File: onnxruntime/core/providers/webnn/builders/impl/cumsum_op_builder.cc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  const auto exclusive = helper.Get("exclusive", 0);
+  const auto reverse = helper.Get("reverse", 0);
+
+  emscripten::val options = emscripten::val::object();
+  options.set("exclusive", exclusive == 1);
+  options.set("reversed", reverse == 1);
+  options.set("label", node.Name());
+
+  emscripten::val output = emscripten::val::object();
+  output = model_builder.GetBuilder().call<emscripten::val>("cumulativeSum", input, narrow<uint32_t>(axis), options);
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/impl/cumsum_op_builder.cc:55]:
```suggestion
  output = model_builder.GetBuilder().call<emscripten::val>("cumulativeSum", input, gsl::narrow<uint32_t>(axis), options);
```
Is `narrow` usually namespaced? I didn't know there was a `using gsl` in ORT.

**Review by fdwr — DISMISSED** [review]
👍

**Comment by fdwr** [comment]
```
/mnt/vss/_work/1/s/onnxruntime/core/providers/webnn/builders/impl/cumsum_op_builder.cc:29:63: error: non-virtual member function marked 'override' hides virtual member function
   29 |                          const logging::Logger& logger) const override;
```

https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1540920&view=logs&j=eeca2222-763b-5799-61a8-571e57814e91&t=75e71cf6-4ecd-5afe-7c9b-28f14f7866d7

**Comment by miaobin** [comment]
> ```
> /mnt/vss/_work/1/s/onnxruntime/core/providers/webnn/builders/impl/cumsum_op_builder.cc:29:63:
> error: non-virtual member function marked 'override' hides virtual member function
>    29 |                          const logging::Logger& logger) const override;
> ```
> 
> https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1540920&view=logs&j=eeca2222-763b-5799-61a8-571e57814e91&t=75e71cf6-4ecd-5afe-7c9b-28f14f7866d7

Fixed and please take another look, thanks a lot!

File: onnxruntime/core/providers/webnn/builders/impl/cumsum_op_builder.cc (hunk 2 above)
Comment (miaobin) [reply onnxruntime/core/providers/webnn/builders/impl/cumsum_op_builder.cc]: Done.

**Review by fdwr — APPROVED** [review]
👍

**Pull request merged**